### ISSUE TITLE
Variable references not detected when used in a spread (in a JSX element)

### ISSUE THREAD

**Issue by myitcv** [Microsoft/TypeScript#4688]
The underlying problem I'm experiencing is described [in this comment](https://github.com/palantir/tslint/issues/570#issuecomment-138332733)

And I think the following test, if my attempts to use the other `tsxRename*.ts` files as a template has succeeded and my understanding of what `tslint` is doing with `getDocumentHighlights` is correct, also highlights the problem:

```typescript
/// <reference path='fourslash.ts' />

//@Filename: file.tsx
//// declare module JSX {
////     interface Element { }
////     interface IntrinsicElements {
////     }
////     interface ElementAttributesProperty { props }
//// }
//// class MyClass {
////   props: {
////     name?: string;
////     size?: number;
////   }
//// }
////
//// var [|nn|]: {name?: string; size?: number};
//// var x = <MyClass {...[|n/**/n|]}></MyClass>;

goTo.marker();
verify.renameLocations(false, false);
```

Does that make sense?

**Comment by RyanCavanaugh** [comment]
Thanks for the test!
